controllers/module_reconciler.go

```diff
@@ -298,6 +316,48 @@ func (r *ModuleReconciler) checkImageExists(ctx context.Context, mod *kmmv1beta1
 	return imageAvailable, nil
 }
 
+func (r *ModuleReconciler) handleDriverContainerServiceAccount(ctx context.Context, mod *kmmv1beta1.Module) error {
+	logger := log.FromContext(ctx)
+
+	sa := &v1.ServiceAccount{}
+	fmt.Println("Before rbacAPI.GetDriverContainerServiceAccount")
+	err := r.rbacAPI.GetDriverContainerServiceAccount(ctx, *mod, sa)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get the driver container service account: %w", err)
+	}
+	fmt.Println("After rbacAPI.GetDriverContainerServiceAccount")
+
+	opRes, err := controllerutil.CreateOrPatch(ctx, r.Client, sa, func() error {
+		return r.rbacAPI.SetDriverContainerServiceAccountAsDesired(ctx, sa, *mod)
+	})
+
+	if err == nil {
+		logger.Info("Reconciled driver container ServiceAccount", "name", sa.Name, "result", opRes)
+	}
+
+	return err
```

Comment:
Nit: could we use a more traditional flow here?
```suggestion
	if err != nil {
		return fmt.Errorf("could not reconcile ServiceAccount: %v", err)
	}
	
	logger.Info("Reconciled driver container ServiceAccount", "name", sa.Name, "result", opRes)
	
	return nil
```

Reply:
Absolutely :+1: